map_test.go

```diff
@@ -1348,6 +1352,13 @@ func TestPerCPUMarshaling(t *testing.T) {
 			}
 
 			// Make sure unmarshaling works on slices containing pointers
+			retrievedVal := make([]*customEncoding, numCPU)
+			for i := range retrievedVal {
+				retrievedVal[i] = &customEncoding{}
```

Comment:
Can you add a case where we pass a a slice which contains a nil pointer and make sure that we don't panic?

Reply:
```
--- FAIL: TestPerCPUMarshaling (0.00s)
    --- FAIL: TestPerCPUMarshaling/PerCPUHash (0.00s)
panic: reflect.Value.Addr of unaddressable value [recovered]
	panic: reflect.Value.Addr of unaddressable value```

haha... I'll add the error handling in the validation.